jni/portolan_sync_api.cpp

```diff
@@ -136,27 +137,23 @@ Java_org_openobservatory_measurement_1kit_jni_sync_PortolanSyncApi_checkPort
         jstring port, jdouble timeout, jboolean verbose) {
     jboolean is_port_open = JNI_FALSE;
     try {
-        Poller poller;
-        Logger custom_logger;
-        custom_logger.set_verbose(verbose);
-        custom_logger.on_log([](const char *s) {
+        Var<Reactor> reactor = Reactor::make();
+        Var<Logger> logger = Logger::make();
+        logger->set_verbosity(verbose);
+        logger->on_log([](uint32_t, const char *s) {
             __android_log_print(ANDROID_LOG_INFO,
                 "portolan-check-port", "%s", s);
         });
-        net::Transport connection = net::connect({
-            {"family", (use_ipv4) ? "PF_INET" : "PF_INET6"},
-            {"address", mk::jni::cxxstring(env, address)},
-            {"port", mk::jni::cxxstring(env, port)},
-        }, &custom_logger, &poller).as_value();
-        connection.set_timeout(timeout);
-        connection.on_error([&poller](Error) {
-            poller.break_loop();
-        });
-        connection.on_connect([&poller, &is_port_open]() {
-            is_port_open = JNI_TRUE;
-            poller.break_loop();
+        reactor->loop_with_initial_event([&]() {
+            net::connect(mk::jni::cxxstring(env, address),
+                         mk::lexical_cast<int>(mk::jni::cxxstring(env, port)),
+                         [&](Error err, Var<mk::net::Transport> txp) {
+                             if (!err) {
+                                 is_port_open = JNI_TRUE;
+                             }
+                             reactor->break_loop();
+                         }, {{"net/timeout", timeout}}, logger, reactor);
```

Comment:
This is basically rewritten to use `loop_with_initial_event()` and the new semantic of `net::connect()`